Code/Tools/AssetProcessor/native/ui/ProductDependencyTreeModel.cpp

```diff
@@ -0,0 +1,363 @@
+        , m_treeType(treeType)
+        , m_fileIcon(QIcon(QStringLiteral(":/AssetProcessor_goto.svg")))
+    {        
+        m_root.reset(new ProductDependencyTreeItem(AZStd::make_shared<ProductDependencyTreeItemData>("", "")));
+    }
+
+    ProductDependencyTreeModel::~ProductDependencyTreeModel()
+    {
+    }
+
+    QModelIndex ProductDependencyTreeModel::index(int row, int column, const QModelIndex& parent) const
+    {
+        if (!hasIndex(row, column, parent))
+        {
+            return QModelIndex();
+        }
+
+        ProductDependencyTreeItem* parentItem = nullptr;
+
+        if (!parent.isValid())
+        {
+            parentItem = m_root.get();
+        }
+        else
+        {
+            parentItem = static_cast<ProductDependencyTreeItem*>(parent.internalPointer());
+        }
+
+        if (!parentItem)
+        {
+            return QModelIndex();
+        }
+
+        ProductDependencyTreeItem* childItem = parentItem->GetChild(row);
+
+        if (childItem)
+        {
+            QModelIndex index = createIndex(row, column, childItem);
+            if (checkIndex(index))
+            {
+                return index;
+            }
+        }
+        return QModelIndex();
+    }
+
+    int ProductDependencyTreeModel::rowCount(const QModelIndex& parent) const
+    {
+        if (parent.column() > 0)
+        {
+            return 0;
+        }
+
+        ProductDependencyTreeItem* parentItem = nullptr;
+        if (!parent.isValid())
+        {
+            parentItem = m_root.get();
+        }
+        else
+        {
+            parentItem = static_cast<ProductDependencyTreeItem*>(parent.internalPointer());
+        }
+
+        if (!parentItem)
+        {
+            return 0;
+        }
+        return parentItem->getChildCount();
+    }
+
+    int ProductDependencyTreeModel::columnCount(const QModelIndex& /*parent*/) const
+    {
+        return static_cast<int>(ProductDependencyTreeColumns::Max);
+    }
+
+    QVariant ProductDependencyTreeModel::data(const QModelIndex& index, int role) const
+    {
+        if (!index.isValid())
+        {
+            return QVariant();
+        }
+
+        ProductDependencyTreeItem* item = static_cast<ProductDependencyTreeItem*>(index.internalPointer());
+        switch (role)
+        {
+        case Qt::DisplayRole:
+            return item->GetDataForColumn(index.column());
+        case Qt::DecorationRole:
+            // Only show the icon in the name column
+            if (index.column() == static_cast<int>(AssetTreeColumns::Name))
+            {
+                return m_fileIcon;
+            }
+            break;
+        case Qt::ToolTipRole:
+            // Purposely return an empty string, so mousing over rows clear out.
+            return QString("");
+        }
+        return QVariant();
+    }
+
+    bool ProductDependencyTreeModel::setData(const QModelIndex& /*index*/, const QVariant& /*value*/, int /*role*/)
```

Comment:
you need to mark these as unused `[[maybe_unused]]`

Reply:
Changed to [[maybe_unused]]